libgloss/arc64/Makefile.in

```diff
@@ -107,15 +123,33 @@ all: $(ALL)
 libcfunc.o: libcfunc.c
 nsim-syscall.o: nsim-syscall.c
 sbrk.o: sbrk.c
+hl-stub.o: hl-stub.c
+hl-setup.o: hl-setup.c
+arc-timer.o: arc-timer.c
+arc-main-helper.o: arc-main-helper.c
 
 $(NSIM_BSP): $(NSIM_OBJS)
 	$(AR) $(ARFLAGS) $@ $?
 	$(RANLIB) $@
 
+$(HL_DIR): hl
+$(HL_DIR):
+	for dir in .. $^; do \
+	  if [ x$$dir != x.. ]; then \
```

Comment:
To be honest I just reused infrastructure from arc32 version https://github.com/foss-for-synopsys-dwc-arc-processors/newlib/blob/4920b468e3be8ce0ace3d18e42bafc685ddf6bd6/libgloss/arc/Makefile.in#L160

So I think we can also keep it as it is, because for arc32 we have two directories to build. What do you think?

Reply:
Fixed.

Reply:
I tried it with this piece of code and it works:

```makefile
$(HL_DIR):
	for dir in $^; do \
	  if [ -d $$dir ]; then \
	    (cd $$dir; $(MAKE) $(FLAGS_TO_PASS) all) || exit $$?; \
	  fi; \
	done
```

The `..` is gone and so is the `if [ x$$dir != x.. ]`.  There is no need for the `else true;` part either.

I've already updated the 32-bit version: [[libgloss/arc] Declutter the Makefile.in](https://github.com/foss-for-synopsys-dwc-arc-processors/newlib/commit/22d463abd)

Reply:
I dropped `else true;`. But please note that this construct is widely used in newlib, just `grep -r 'else true' newlib/`. Maybe some shells process non-elsed ifs differently, maybe it is MISRA shell :)

Reply:
That is a good point indeed! I tried again the changes inside [`dash`](http://gondor.apana.org.au/~herbert/dash/) and it was OK. If indeed it goes wrong somewhere (_MISRA_ or not), we'll know later. During the upstream we will know for sure.